File: BenchmarkDotNet.IntegrationTests/PretendFSharpTest.cs
using BenchmarkDotNet.Configs;
using BenchmarkDotNet.Loggers;
using BenchmarkDotNet.Running;
using Xunit;

namespace BenchmarkDotNet.IntegrationTests
{
    // The referenced file "FSharpBenchmarkDotNet.exe" is compiled from F# code as shown in the file "PretendFSharpTest.fs.
    // That .fs file is NOT meant to be compiled as part of our Integration tests, it is just there for reference.
    // (It seemed simpler to do it this way, rather than trying to compile F# code as part of our integration tests)
    public class PretendFSharpTest
    {
        [Fact]
        public void Test()
        {
            var logger = new AccumulationLogger();
            var config = DefaultConfig.Instance.With(logger);
            // Run our "Pretend F# test" (see above for more info)
            BenchmarkRunner.Run<BenchmarkSpec.Db>(config);
            var testLog = logger.GetLog();
            Assert.Contains("// ### F# Benchmark method called ###", testLog);
            Assert.DoesNotContain("No benchmarks found", testLog); // TODO: move message to const for all of the test
        }
    }
}

File: BenchmarkDotNet.IntegrationTests/PretendFSharpTest.fs
// This file is NOT meant to be compiled as part of our Integration Tests
// It is just here for reference, see PretendFSharpTest.CS for more information
module BenchmarkSpec

open BenchmarkDotNet.Attributes
open NUnit.Framework
open FsUnit
open System.Threading

type File = 
    { Name : string
      Path : string
      Extension : string
      Length : int }

[<Config("jobs=dry")>]
type Db() = 

    let createDoc name = 
        { Name = name
          Path = name
          Extension = name
          Length = name.Length }

    [<Benchmark>]
    member this.Test() = 
        printfn "// ### F# Benchmark method called ###"
        Thread.Sleep(50)
        createDoc("Testing")

[<Test>]
let ShouldExecuteBenchmark() = 
    let reports = BenchmarkRunner().Run<Db>()
    ()


// Can't get NUnit test runner to work in VS, so "simulate" it by calling the Test method from the EntryPoint method (main())
[<EntryPoint>]
let main argv = 
    printfn "Running NUnit Test Method: ShouldExecuteBenchmark()"
    ShouldExecuteBenchmark()
    0 // return an integer exit code